### ISSUE TITLE
[a11y] Exit button doesn't announce self

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/firefox-tv#2484]
### Vision statement / What / Requirements
Currently the contentDescription is null:
https://github.com/mozilla-mobile/firefox-tv/blob/202345441ebdd778066efd9dae92c66cf68eb9fc/app/src/main/res/layout/fragment_navigation_overlay_top_nav.xml#L55

Presumably (I haven't tested on device) this means this button is not being announced for accessibility.

### Impact
Screen readers users will most likely not recognize the exit button.

### Acceptance criteria
- [ ] Add content description for exit button (note: Eitan said any description is better than none so it's probably not worth waiting for Bram or others for "proper copy": we should just make it ourselves)
